libslide/src/evaluator_rules/pattern_matching.rs

```diff
@@ -102,6 +119,7 @@ impl Replacements {
                 }
```

Comment:
We shouldn't just return none. I think a nice way to do it is return a Result < self, self::error> that way we mirror the try_from logic rust already mandates.

Reply:
Why not? What would the error be? To me, an Option equivalent to  `Result<Self, ()>` where the error is implicit - processing failed.

Reply:
I think its better code to make errors explicit for anyone who looks or uses our library. Implicit errors can be impossible to find and theres certainly going to be bugs.

Reply:
In general I agree with you, but in this method we only fail in one place that is documented on the function definition. Furthermore, (1) this method is internal to libslide and will not be available publicly, and (2) no one would consume the result of the error even if we did provide it, because it is not propogated up if we fail to match a rule -- we just skip it and move on.

Reply:
will try_merge fail often then if we are matching multiple rules.

Reply:
I am not sure I understand the point of this method and how it will be used.

Reply:
Yes, but separately each time. It is meant to merge two sets of replacements. Basically the idea is let's say I have a binary expression "1 + 2" that I want to match against "#a + #a". Matching each LHS of the target expression against each LHS of the rule will work, and I will get two sets of replacements: "{#a: 1}" and "{#a: 2}". The problem is, this is not actually valid across the entire addition expression, because 1 != 2. So when determining the replacements for the entire addition expression, we have to merge these two replacements sets, at which point we determine that they are invalid and therefore there is no matching pattern.

Reply:
i got it. So we are basically applying rules where the terms may not neccesarily be the same. Thanks for explaining. we don't need an error.